src/main/java/com/vinylteam/vinyl/util/impl/VinylUaParser.java

```diff
@@ -4,141 +4,241 @@
 import com.vinylteam.vinyl.entity.RawOffer;
 import com.vinylteam.vinyl.util.VinylParser;
 import org.jsoup.Jsoup;
+import org.jsoup.nodes.DataNode;
 import org.jsoup.nodes.Document;
 import org.jsoup.nodes.Element;
 import org.jsoup.select.Elements;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.LinkedHashSet;
-import java.util.List;
+import java.util.*;
 
 public class VinylUaParser implements VinylParser {
 
     private final Logger logger = LoggerFactory.getLogger(getClass());
-    private final String startLink = "http://vinyl.ua";
-    private final String classContainingGenresLinks = "dropdown-menu dropdown-menu-left";
-    private final String classContainingPagesLinks = "pagination-wrapper margin-top-20";
-    private final String classContainingVinyl = "vinyl-release showcase";
-    private final String classContainingVinylRelease = "margin-top-clear margin-bot-5";
-    private final String classContainingVinylArtist = "text-ellipsis";
-    private final String classContainingVinylPriceDetails = "pull-left margin-top-5 showcase-release-price";
-    private final String classContainingVinylLinks = "img-showcase-release";
+    private final String START_LINK = "http://vinyl.ua";
+    private final String SELECTOR_SCRIPT_WITH_HIGH_RES_IMAGE_LINK = "script:containsData(openPhotoSwipe)";
+    private final String SELECTOR_RELEASE = "div.boxed div.col-sm-7 > h3.normal-text";
+    private final String SELECTOR_ARTIST = "div.boxed div.col-sm-7 > h4.normal-text > span.text-ellipsis > a";
+    private final String SELECTOR_PRICE_DETAILS = "div.boxed div.col-sm-4 > button.btn-success > b";
+    private final String SELECTOR_GENRE = "div.boxed div.col-xs-12 table.list-meta td.text-right:contains(Жанр:) + td";
+    private final String SELECTOR_GENRE_ANCHORS = "nav#intro div#bs-example-navbar-collapse-1 > ul.nav > li.dropdown > ul.dropdown-menu > li > a";
+    private final String SELECTOR_PAGE_ANCHORS = "div.pagination-wrapper > ul.pagination > li:not(.nav-pagi) > a";
+    private final String SELECTOR_OFFER_ANCHORS = "div.row > div.col-sm-9 > div.row div.vinyl-release > div.boxed > p > a";
```

Comment:
You can use upper case for naming variable if it is constant.
private final String START_LINK - it is not constant because for each object of this class it will separate instance of this variable.
Please make all of them static